### ISSUE TITLE
BUG: Text detaching from figures when dragging them (reopening #538)

### ISSUE THREAD

**Issue by Ranastra-wolf** [zsviczian/obsidian-excalidraw-plugin#1400]
**Describe the bug**
The bug is identical to the one described in #538. When dragging a figure, the text of that figure remains in the original location. Most of the time the figure will be empty, but I also had a few occations where the text duplicated.

However, most of the time this issue can be resolved by clicking (several times) on the text. At some point the text will move to the intended location where the figure is. Usually this will be by first moving to the correct position on the y-axis and after clicking one more time it will also move to the correct position on the x-axis.

Also, when deleting the figure, the text will be deletedaswell, so they are still connected.

It seems to me, but I am no experienced programmer, that the text item should update automatically, but now it only updates when I click on it.

For instance, here it happend with an entire graph I copy pasted from another file:
![image](https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/149255880/2b2baf61-3caa-4040-88f6-37903bff082a)

When I save it and re-open it it looks normal, except the text is duplicated:
![image](https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/149255880/3b52dd3c-4268-4a65-897e-4296a157dc1a)

Except after dragging a figure and then pressing ctrl-z. Then the figure is empty again, but the text that was on the right moves together with the figures on the left when I drag them (which I did here with 'Algorithm')
![image](https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/149255880/4aa35735-d2bd-4909-a157-6b5fdc595592)

And when I click on the text on the right, it will be placed in the figure it belongs to. First to the correct position on the y-axis (which I can't show because when creating a partial prtscr, the folllowing laready happens), then after clicking one more time (this time it can be anywhere) to the correct position on the x axis aswell:
![image](https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/149255880/d0e1703a-736c-4fed-a81f-6e0346dcef23)

From here on the figure and text move normal. Dragging the figure will also drag the text. 

To be clear, this behaviour also happens within a file, not only after copy-pasting. But this is an example I just encountered. Also, as I was able to reproduce the problem after saving the file, I figured adding the file here would maybe help you to reproduce the problem.
[Bug_example.excalidraw.md](https://github.com/zsviczian/obsidian-excalidraw-plugin/files/13196221/Bug_example.excalidraw.md)

**To Reproduce**
Select a figure with text and drag it. However, it's not clear when it happens. I feel like it happens mostly after copy-pasting a figure (within one file or across excalidraw files), but I haven´t been keeping track to know whether this is actually true.

It looks somewhat similar to the function where you select text in a figure and drag this text, which creates a new text item. In this case it is intentionally, but when this bug happens it is definitly not.

**Expected behavior**
To drag the text with the figure.

I hope that this bit of additional information can help you to figure out what is the cause of this bug. Either way, I'm very happy with exalidraw :)

**Comment by zsviczian** [comment]
I am unable to reproduce. However, based on the file you provided (very helpful!) I created a partial fix that will detach the secondary text element when saving/loading the drawing. CTRL+S should force it to perform.

This is not a fix, but only sticking plaster.
Until I am able to reproduce the issue from scratch... I can't debug what is happening but only make assumptions.

**Comment by awmottaz** [comment]
I ran into a similar bug, and I can reliably reproduce it using the frame tool.


https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/17735385/37f17095-8609-46b0-9e64-71f4839d0823

Here's another video where things continue to look broken:


https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/17735385/3195a0df-46b7-4a96-8cac-6cb42ceed85e

I'm using:

- Obsidian v1.5.2
- Excalidraw v2.0.11

You can find the markdown file (copied at the conclusion of the second video) and my `data.json` in this Gist: https://gist.github.com/awmottaz/d8c17681ec70458870cdd119006a20c3

Let me know if I can provide any more info!

**Comment by zsviczian** [comment]
I am closing this as I've not seen this happen for a long time now. If the issue is still there, we can reopen, or create a new issue.
